### PR TITLE
chore(l2): forbid expects

### LINKED ISSUE #1267: L2: Forbid `expect`s
Add the following clippy lint to all `Cargo.toml` files in the `l2` crate:

```toml
[lints.clippy]
expect = "deny"
```

and remove the `expect`s of the module.

### PR DESCRIPTION
**Pull request by dsocolobsky**
**Motivation**

We should not be using `.expect()` and use Errors instead.

**Description**

* `expect_used = "deny"` was added to `Cargo.toml` to throw clippy errors on expects.
* `.expect()` usage was replaced with errors

Closes  #1267

### CODE REVIEW

**Review by fborello-lambda — CHANGES_REQUESTED** [review]
If we run the stack with the following steps:
- `cd crates/l2`
- `cp .env.example .env`
- `make init`
- do `ctrl-c` to stop the server, in order to restart it without some env variables.
- Comment some env variables starting with `COMMITTER`, `L1_WATCHER`, `PROPOSER`... etc
- `make init-l2`

It will not throw an error nor a message saying that there are missing variables. We should throw an error message and even stop the execution saying that some config variables are missing.

File: crates/l2/proposer/l1_watcher.rs
Comment (dsocolobsky) [reply crates/l2/proposer/l1_watcher.rs:21]: Do you mean I should panic! here in each function or bubble up to the `try_join` and do it there?

File: crates/l2/proposer/mod.rs
Comment (dsocolobsky) [reply crates/l2/proposer/mod.rs:35]: I thought this was checking the error, I'm not familiar with tokio. What can I do to check it? Or do you mean I should panic! instead of error! ?

Comment (fborello-lambda) [reply crates/l2/proposer/mod.rs]: the JoinSet you included works fine

File: crates/l2/prover/zkvm/interface/guest/src/main.rs
Comment (fborello-lambda) [inline crates/l2/prover/zkvm/interface/guest/src/main.rs:38]: ditto

File: crates/l2/contracts/deployer.rs
Comment (fborello-lambda) [inline crates/l2/contracts/deployer.rs:42]: I've commented in https://github.com/lambdaclass/ethrex/pull/1290#discussion_r1861096826 . We can use a struct instead of a type

**Review by fborello-lambda — APPROVED** [review]
LGTM, ready to merge if the CI passes. Just make sure that `type SetupResult` is a `struct`. in PR #1290 or in this one.

File: crates/l2/prover/zkvm/interface/guest/Cargo.toml
Comment (dsocolobsky) [reply crates/l2/prover/zkvm/interface/guest/Cargo.toml]: Should be fixed in 4e99d9ba79ec33aac1e0cd299a39ed2a9c2737e5

File: crates/l2/prover/zkvm/interface/guest/src/main.rs
Comment (dsocolobsky) [reply crates/l2/prover/zkvm/interface/guest/src/main.rs]: Fixed in 4e99d9ba79ec33aac1e0cd299a39ed2a9c2737e5

File: crates/l2/prover/zkvm/interface/guest/src/main.rs
Comment (dsocolobsky) [reply crates/l2/prover/zkvm/interface/guest/src/main.rs]: Fixed in 4e99d9ba79ec33aac1e0cd299a39ed2a9c2737e5

File: crates/l2/prover/zkvm/interface/guest/src/main.rs
Comment (dsocolobsky) [reply crates/l2/prover/zkvm/interface/guest/src/main.rs]: Fixed in 4e99d9ba79ec33aac1e0cd299a39ed2a9c2737e5

File: crates/l2/prover/zkvm/interface/guest/src/main.rs
Comment (dsocolobsky) [reply crates/l2/prover/zkvm/interface/guest/src/main.rs]: Fixed in 4e99d9ba79ec33aac1e0cd299a39ed2a9c2737e5

File: crates/l2/prover/zkvm/interface/guest/src/main.rs
Comment (dsocolobsky) [reply crates/l2/prover/zkvm/interface/guest/src/main.rs]: Fixed in 4e99d9ba79ec33aac1e0cd299a39ed2a9c2737e5

File: crates/l2/contracts/deployer.rs
Comment (dsocolobsky) [reply crates/l2/contracts/deployer.rs]: I changed it in a77da2c744bd769766df3c3f5a364527fc105db3 so that it uses the same struct as in #1290

**Review by ilitteri — APPROVED** [review]

**Pull request merged**
